### DISCUSSION TITLE
Context Propagation in Java + Open Telemetry

### DISCUSSION THREAD

**Discussion by shameekagarwal** [open-telemetry/opentelemetry-specification#3515] [Q&A]
Hi!
To get started with opentelemetry for java based applications, all I had to do is [download the java agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation#getting-started), and it was able to send tracing data to jaeger automatically (impressive!).  
I can also see the [trace id and span ids in logs](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/logger-mdc-instrumentation.md).  
However, I do not see the trace data in the response headers. This makes me think is context propagation not enabled automatically?  
How can i add context propagation with automatic instrumentation?  
_PS - my java applications can range from drop wizard to spring boot to camel, with no consistency..._

**Comment by trask** [comment]
hi @shameekagarwal, I'd recommend asking this in the [OpenTelemetry Java Instrumentation repository](https://github.com/open-telemetry/opentelemetry-java-instrumentation)
